comp/logs/agent/config/endpoints.go

```diff
@@ -238,6 +267,52 @@ func (e *Endpoint) GetStatus(prefix string, useHTTP bool) string {
 	return fmt.Sprintf("%sSending %s logs in %s to %s on port %d", prefix, compression, protocol, host, port)
 }
 
+// onConfigUpdate handles configuration change notification to update the internal API key of the Endpoint if needed
+func (e *Endpoint) onConfigUpdate(l *LogsConfigKeys) {
+	l.getConfig().OnUpdate(func(key string, oldVal interface{}, newVal interface{}) {
+		if key != e.configSettingPath {
+			return
+		}
+
+		// main Endpoints can directly get their API key from the configuration without having to load complex
+		// types.
+		if !e.isAdditionalEndpoint {
+			if newAPIKey, ok := newVal.(string); !ok {
```

Comment:
Could this check be pulled up, outside of the if block for `e.isAdditionalEndpoint`? Or is `newVal` only valid for the main endpoint and invalid fro additional endpoints?

Reply:
We don't rely on `newVal` for additional endpoints as we use the helper `getAdditionalEndpoints` (`newVal` is a  `[]map[interface {}]interface{}` and the helper already parse it).